### PR TITLE
Support OpenVINO ResNet50-int8 model with Inference Model

### PR DESCRIPTION
**Pull request by qiyuangong**

### CODE REVIEW

**Comment by qiyuangong** [comment]
Jenkins passed http://10.239.47.211:18888/job/ZOO-PR-Validation/2255/ .

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Analytics Zoo OpenVINO Int8 ResNet_v1_50 Inference Example
+Based on [analytics-zoo](https://github.com/intel-analytics/analytics-zoo) and [OpenVINO](https://software.intel.com/en-us/openvino-toolkit).
+
+## Summary
+
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/README.md]: ???

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+* Apache Spark (This version needs to be same with the version you use to build Analytics Zoo)
+* [Analytics Zoo](https://analytics-zoo.github.io/master/#ScalaUserGuide/install/)
+
+## Datasets and pre-trained models
+* Datasets: [ImageNet2012 Val](http://image-net.org/challenges/LSVRC/2012/index)
+* Pre-trained model: [ResNet50 for OpenVINO]()
+* Int8 Optimized model: [ResNet50 Int8 for OpenVINO]()
+
+
+
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/README.md]: Running commands?

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/ImageNetEvaluation.scala [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      opt[String]('f', "folder")
+        .text("The path to the imagenet dataset for inference")
+        .action((x, c) => c.copy(folder = x))
+        .required()
+      opt[String]('m', "model")
+        .text("The path to the int8 quantized ResNet50 model snapshot")
+        .action((x, c) => c.copy(model = x))
+        .required()
+      opt[String]('w', "weight")
+        .text("The path to the int8 ResNet50 model weight")
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/ImageNetEvaluation.scala]: A little bit confusing in description here? Here you say the path to int8 model and weight; and you have another argument isInt8? So both int8 and fp32 models are supported in this example?

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/ImageNetEvaluation.scala [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        .action((x, c) => c.copy(batchSize = x))
+      opt[Int]("partitionNum")
+        .text("The partition number of the dataset")
+        .action((x, c) => c.copy(partitionNum = x))
+      opt[Boolean]("isInt8")
+        .text("Is Int8 optimized model?")
+        .action((x, c) => c.copy(isInt8 = x))
+    }
+    parser.parse(args, ImageNetEvaluationParams()).foreach(param => {
+      val sc = NNContext.initNNContext("ImageNet2012 with OpenVINO Inference Example")
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/ImageNetEvaluation.scala]: Inference -> Evaluation

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/ImageNetEvaluation.scala [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+      }
+      val batchNum = results.count().toFloat
+      val timeUsed = System.nanoTime() - start
+      // Post-processing
+      val throughput = "%.2f".format(imageNum / (timeUsed / 1e9))
+      val batchLatency = "%.2f".format(timeUsed / 1e6 / batchNum)
+      logger.info(s"Takes $timeUsed ns, throughput is $throughput FPS (imgs/sec)")
+      logger.info(s"Average Predict latency is $batchLatency ms")
+      // Evaluation
+      // Compare labels and output results
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/ImageNetEvaluation.scala]: are we doing the comparision? Is this an evaluation or a predict example?

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/Perf.scala [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        model.doLoadOpenVINO(param.model, param.weight)
+      }
+
+      /*
+      val predictStart = System.nanoTime()
+      val threads = List.range(0, param.iteration).map(_ => {
+        new Thread() {
+          override def run(): Unit = {
+            model.doPredict(batchInput)
+          }
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/Perf.scala]: are these comments used?

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/README.md (hunk 2 above)
Comment (hkvision) [reply zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/README.md]: Especially where can users download models?

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/VINOPerf.scala [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+import java.util.Arrays
+
+import com.intel.analytics.zoo.pipeline.inference.{InferenceModel, JTensor}
+import org.apache.log4j.Logger
+import scopt.OptionParser
+
+import scala.util.Random
+
+
+object VINOPerf {
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/VINOPerf.scala]: What's the difference between Perf and VINOPerf?

Comment (qiyuangong) [reply zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/VINOPerf.scala]: VINOPerf depends only on InferenceModel and JTensor, while Perf depends on Zoo and bigDL.

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/Perf.scala (hunk 6 above)
Comment (qiyuangong) [reply zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/Perf.scala]: Will remove. :)

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/ImageNetEvaluation.scala (hunk 3 above)
Comment (qiyuangong) [reply zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/ImageNetEvaluation.scala]: Yes. Both fp32 and uint8 is supported.  The only problem is that user should figure out if model is fp32 or uint8.

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/ImageNetEvaluation.scala (hunk 5 above)
Comment (qiyuangong) [reply zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/ImageNetEvaluation.scala]: Thanks. Already added. :)

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/README.md (hunk 1 above)
Comment (qiyuangong) [reply zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/README.md]: Document added. :)

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/README.md (hunk 2 above)
Comment (qiyuangong) [reply zoo/src/main/scala/com/intel/analytics/zoo/examples/vnni/openvino/README.md]: Download link will be added after we release zoo-core and optimized model.

**Pull request merged**